app/src/main/kotlin/com/wire/android/ui/home/conversations/details/editguestaccess/createPasswordProtectedGuestLink/CreatePasswordProtectedGuestLinkScreen.kt

```diff
@@ -0,0 +1,258 @@
+    navigator: Navigator,
+    viewModel: CreatePasswordGuestLinkViewModel = hiltViewModel(),
+) {
+    val scrollState = rememberScrollState()
+    val snackbarHostState = remember { SnackbarHostState() }
+
+    val clipboardManager = LocalClipboardManager.current
+    val onCopyClick = remember(viewModel.state.password.text) {
+        {
+            if (viewModel.state.isPasswordValid) {
+                clipboardManager.setText(viewModel.state.password.annotatedString)
+                viewModel.onPasswordCopied()
+            }
+        }
+    }
+    if (viewModel.state.isLinkCreationSuccessful) {
+        if (viewModel.state.isPasswordCopied) {
+            navigator.navigateBack()
+        } else {
+            PasswordNotCopiedDialog {
+                onCopyClick()
+                navigator.navigateBack()
+            }
+        }
+    }
+
+    Scaffold(topBar = {
+        WireCenterAlignedTopAppBar(
+            elevation = scrollState.rememberTopBarElevationState().value,
+            onNavigationPressed = navigator::navigateBack,
+            title = stringResource(id = R.string.conversation_options_create_password_protected_guest_link_title),
+        )
+    }, snackbarHost = {
+        SwipeDismissSnackbarHost(
+            hostState = snackbarHostState, modifier = Modifier.fillMaxWidth()
+        )
+    }) { internalPadding ->
+        Column {
+            LazyColumn(
+                modifier = Modifier
+                    .background(MaterialTheme.wireColorScheme.background)
+                    .padding(internalPadding)
+                    .padding(
+                        start = dimensions().spacing16x,
+                        end = dimensions().spacing16x,
+                        bottom = dimensions().spacing16x,
+                        top = dimensions().spacing16x,
+                    )
+                    .weight(1F)
+                    .fillMaxSize()
+            ) {
+
+                item {
+                    Text(
+                        text = stringResource(
+                            id = R.string.conversation_options_create_password_protected_guest_link_discrption
+                        ),
+                        style = MaterialTheme.wireTypography.body02.copy(fontWeight = FontWeight.Normal)
+                    )
+                    Spacer(modifier = Modifier.height(16.dp))
+                    Text(
+                        text = stringResource(
+                            id = R.string.conversation_options_create_password_protected_guest_link_discrption_2
+                        ),
+                        style = MaterialTheme.wireTypography.body02
+                    )
+                    Spacer(modifier = Modifier.height(24.dp))
+                }
+
+                item {
+                    GeneratePasswordButton(
+                        onClick = viewModel::onGenerateRandomPassword
+                    )
+                    Spacer(modifier = Modifier.height(24.dp))
```

Comment:
Question: why don't using the dimensions?

Reply:
this is repeated in many places, wondering is something you're working on, or there is a reason behind it.